apollo-router/src/services/fetch_service.rs

```diff
@@ -56,30 +119,30 @@ impl tower::Service<FetchRequest> for FetchService {
         } = request;
 
         let FetchNode {
+            service_name,
             operation,
             operation_kind,
             operation_name,
-            service_name,
             requires,
             output_rewrites,
             id,
             ..
         } = fetch_node;
 
-        let service_name_string = service_name.to_string();
-
-        // TODO: perf
+        // TODO: remove this when older private preview and source-aware code is removed
         let (service_name, subgraph_service_name) = match &*fetch_node.protocol {
```

Comment:
this should be easy to remove — there's no difference between `service_name` and `subgraph_service_name` for GraphQL subgraph requests

Reply:
This was just to keep existing tests running, I think they are using the older directives. Once those are removed, this can be too.

Reply:
ah yeah i see the old nullability_formatting test — let's delete that since it's in the old services file. we can test that same behavior in another way